aderyn_core/src/detect/detector.rs

```diff
@@ -53,6 +57,54 @@ pub fn get_all_detectors() -> Vec<Box<dyn Detector>> {
     ]
 }
 
+pub fn get_all_detectors_names() -> Vec<String> {
+    vec![
+        "delegate-call-in-loop".to_string(),
+        "centralization-risk".to_string(),
+        "solmate-safe-transfer-lib".to_string(),
+        "avoid-abi-encode-packed".to_string(),
+        "ercrecover".to_string(),
+        "deprecated-oz-functions".to_string(),
+        "unsafe-erc20-functions".to_string(),
+        "unspecific-solidity-pragma".to_string(),
+        "zero-address-check".to_string(),
+        "useless-public-function".to_string(),
+        "constants-instead-of-literals".to_string(),
+        "unindexed-events".to_string(),
+        "require-with-string".to_string(),
+        "non-reentrant-before-others".to_string(),
+        "block-timestamp-deadline".to_string(),
+        "unsafe-erc721-mint".to_string(),
+        "push-zero-opcode".to_string(),
+        "arbitrary-transfer-from".to_string(),
+    ]
+}
+
+pub fn get_detector_by_name(detector_name: &str) -> Box<dyn Detector> {
```

Comment:
We must alter these two functions each time we add/remove a dtector implementation. Is there a way in which we can achieve the same functionality without that requirement?

Reply:
Now we have Enums that can derive kebab-case strings : ) 

(The only reason I'm not including hashmap is because I'd then have to force the bot writers to make their detector structs Cloneable because how dyn-traits work :P )